benchmark/C/Savina/src/parallelism/GuidedSearch.lf

```diff
@@ -0,0 +1,246 @@
+            self->nodesToVisit.pop_front();
+            SET(search[workerIndex], WorkMessage(startNode, self->grid->targetNode()));
+            workerIndex++;
+        }
+        
+         // continue with the next iteration
+         schedule(next, 0);
+    =}
+    
+    reaction(pathFound) -> finished {=
+        self->pathWasFound = true;
+        bool valid = self->grid->validate();
+        info_print("Found a path through the grid");
+        info_print("Result valid: %d", valid);
+        SET(finished, true);
+    =}
+    
+    reaction (moreNodesToVisit) {=
+        // collect more nodes to visit from all workers and store them in the internal queue
+        for (int i=0; i< moreNodesToVisit_width; i++) {
+            if (moreNodesToVisit[i]->is_present) {
+                auto nodes = moreNodesToVisit[i]->value;
+                if (!nodes.empty()) {
+                    self->nodesToVisit.insert(self->nodesToVisit.end(), nodes.begin(), nodes.end());
+                }
+            }
+        }
+    =}
+}
+
+reactor Worker(bank_index: int(0), threshold: size_t(1024)) {
+    
+    input search: WorkMessage;
+    output pathFound: bool;
+    output moreNodesToVisit: std::deque<GridNode*>;
+
+    preamble {=
+
+        void busyWait() {
+            for (int i = 0; i < 100; i++) {
+                rand();
+            }
+        }
+    =}
+    
+    reaction(search) -> pathFound, moreNodesToVisit {=
+        auto workMessage = search->value;
+        GridNode* targetNode = workMessage.target;
+        GridNode* currentNode = workMessage.node;
+
+        char current_node_str_rep[100];
+        char target_node_str_rep[100];
+        info_print("Worker %d: search path from %s to %s.", self->bank_index, currentNode->toString(current_node_str_rep, 100), targetNode->toString(target_node_str_rep, 100));
+        auto workQueue = std::deque<GridNode*>();
+        
+        workQueue.push_back(currentNode);
+        
+        // start the search, but visit at most threshold nodes
+        size_t nodesProcessed{0};
+        while(!workQueue.empty() && nodesProcessed < self->threshold) {
+            
+            nodesProcessed++;
+            busyWait();
+            
+            GridNode* loopNode = workQueue.front();
+            workQueue.pop_front();
+            size_t numNeighbors = loopNode->numNeighbors();
+            
+            for (size_t i{0}; i < numNeighbors; i++) {
+                GridNode* loopNeighbor = loopNode->neighbor(i);
+                bool success = loopNeighbor->setParent(loopNode);
+                if(success) {
+                    if(loopNeighbor == targetNode) {
+                        SET(pathFound, true);
+                        return;
+                    } else {
+                        workQueue.push_back(loopNeighbor);
+                    }
+                }
+            }
+        }
+        
+        if(!workQueue.empty()) {
+            SET(moreNodesToVisit, std::move(workQueue));
```

Comment:
I am not sure how the `std::move` plays along here with the `SET` macro. I think you are good here and this will only copy the deque object to the port value, but not copy its data.